src/coreComponents/physicsSolvers/fluidFlow/FlowSolverBase.cpp

```diff
@@ -712,5 +712,58 @@ void FlowSolverBase::saveAquiferConvergedState( real64 const & time,
   } );
 }
 
+void FlowSolverBase::prepareStencilWeights( DomainPartition & domain )
+{
+  forDiscretizationOnMeshTargets( domain.getMeshBodies(), [&] ( string const &,
+                                                                MeshLevel & mesh,
+                                                                arrayView1d< string const > const & regionNames )
+  {
+    GEOS_UNUSED_VAR( regionNames );
+
+    NumericalMethodsManager const & numericalMethodManager = domain.getNumericalMethodManager();
+    FiniteVolumeManager const & fvManager = numericalMethodManager.getFiniteVolumeManager();
+    FluxApproximationBase const & fluxApprox = fvManager.getFluxApproximation( getDiscretizationName() );
+    ElementRegionManager::ElementViewAccessor< arrayView1d< real64 const > > hydraulicAperture =
+      mesh.getElemManager().constructViewAccessor< array1d< real64 >, arrayView1d< real64 const > >( fields::flow::hydraulicAperture::key() );
+    auto hydraulicApertureView = hydraulicAperture.toNestedViewConst();
+
+    fluxApprox.forStencils< SurfaceElementStencil, FaceElementToCellStencil, EmbeddedSurfaceToCellStencil >( mesh, [&]( auto & stencil )
+    {
+      typename TYPEOFREF( stencil ) ::KernelWrapper stencilWrapper = stencil.createKernelWrapper();
+
+      forAll< parallelDevicePolicy<> >( stencilWrapper.size(), [=] ( localIndex const iconn )
```

Comment:
looking at the error I think the fix is actually quite simple. 

```suggestion
      forAll< parallelDevicePolicy<> >( stencilWrapper.size(), [=] GEOS_HOST_DEVICE ( localIndex const iconn )
```